### PR TITLE
make encryption stream wrapper seekable

### PR DESCRIPTION
**Pull request by jknockaert**
OC8 version of #12909.

### CODE REVIEW

**Comment by MorrisJobke** [comment]
cc @schiesbn

File: 3rdparty [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 5142d69c5c467c651a7ef72ea1f09dcfb7ba25b5
+Subproject commit eb12e0f6e04cc17bae34a4971349d9b3578b21ad
```
Comment (LukasReschke) [inline 3rdparty]: That looks unwanted?

**Comment by LukasReschke** [comment]
@owncloud-bot This is okay to test.

File: 3rdparty (hunk 1 above)
Comment (jknockaert) [inline 3rdparty]: yes, couldn't fix that one yet

**Comment by schiesbn** [comment]
@th3fallen please also review this PR. We need to integrate this in our work as well.

**Comment by jknockaert** [comment]
@LukasReschke I fixed the unwanted 3rdparty diff

**Comment by jknockaert** [comment]
Is anyone reviewing this PR? Pls let me know if there's anything that needs to be finished before review can start.

**Comment by MorrisJobke** [comment]
@jknockaert A license statement would be nice ;)

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

**Comment by jknockaert** [comment]
@MorrisJobke I sent in a signed contributor agreement on 28 April 2014. Is there a separate statement to be made for each PR?

**Comment by MorrisJobke** [comment]
@jknockaert Ah okay. Nothing needed then.

@karlitschek @DeepDiver1975 Why is the bot then so confused? Isn't @jknockaert in the owncloud/dev group?

**Comment by karlitschek** [comment]
@jknockaert Hmm. Strange. I don´t have the agreement here. You email a photo of it to frank at owncloud.org please?
Sorry for the trouble.

**Comment by jknockaert** [comment]
@karlitschek Done.

**Comment by karlitschek** [comment]
@jknockaert Great. Please confirm the github email invite and you have access. Thanks a lot and sorry for the delay :-)

**Comment by MorrisJobke** [comment]
@schiesbn @th3fallen check this out and review it :)

**Comment by schiesbn** [comment]
I gave it a short try and I'm fine with merging it :+1: 

@jknockaert thanks again for all your work and patience.

(I already start to porting part of it to the new encryption)

**Comment by jknockaert** [comment]
OK. I will wait for @th3fallen before merging.

**Comment by th3fallen** [comment]
fine by me @jknockaert :+1:

**Comment by jknockaert** [comment]
I created #15113 to revert #10422 to allow http-range requests on encrypted files.

**Pull request merged**